src/libflate/lib.rs

```diff
@@ -35,13 +34,27 @@
 extern crate libc;
 
 use libc::{c_void, size_t, c_int};
+use std::fmt;
 use std::ops::Deref;
 use std::ptr::Unique;
 use std::slice;
 
+#[derive(Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Error(());
+
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        "decompression error".fmt(f)
+    }
+}
+
+impl Drop for Error {
```

Comment:
Why is `Error` defined as a tuple-struct in the first place? I think this is generally not a good idea (#22045). Implementing a pointless `Drop` seems like a bad solution.